src/nvim/main.c

```diff
@@ -328,6 +326,9 @@ int main(int argc, char **argv)
                  "'\\c\\m" PROTO "\\(.\\{-}\\)//'), 1, '')})");
 #undef PROTO
 
+  do_cmdline_cmd("autocmd PastePre  * set paste|if mode()!~#'c\\|t'|call feedkeys('\x1C\xEi','n')|endif");
+  do_cmdline_cmd("autocmd PastePost * set nopaste|if mode()!~#'c\\|t'|call feedkeys('\x1C\xE','n')|endif");
```

Comment:
And I would also suggest to create autocmd group `_Builtin` for all built-in autocommands (currently only this one and term://\* BufReadCmd), so that they are easy to identify, making user aware of what he sees and, possibly, changes.